source/extensions/filters/http/tap/config.cc

```diff
@@ -15,21 +15,28 @@ namespace TapFilter {
 
 class HttpTapConfigFactoryImpl : public Extensions::Common::Tap::TapConfigFactory {
 public:
+  HttpTapConfigFactoryImpl(TimeSource& time_source) : time_source_(time_source) {}
+
   // TapConfigFactory
   Extensions::Common::Tap::TapConfigSharedPtr
   createConfigFromProto(const envoy::config::tap::v3::TapConfig& proto_config,
                         Extensions::Common::Tap::Sink* admin_streamer) override {
-    return std::make_shared<HttpTapConfigImpl>(std::move(proto_config), admin_streamer);
+    return std::make_shared<HttpTapConfigImpl>(std::move(proto_config), admin_streamer,
+                                               time_source_);
   }
+
+private:
+  TimeSource& time_source_;
```

Comment:
Why does the factory need to store the `time_source_`?
I think it may make more sense to move to the FilterConfigImpl.

Reply:
All those code Used from transportSocket tap.
from my understanding, it needs pass it for line 24.

Reply:
I'm not saying there's no need to pass the time-source for a specific filter. But why is it needed as part of the state of the factory?
It should probably be part of the `FilterConfigImpl` instead.